<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.2.0">


  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">

<link rel="stylesheet" href="//fonts.googleapis.com/css?family=Palatino:300,300italic,400,400italic,700,700italic|Libre+Baskerville:300,300italic,400,400italic,700,700italic|Roboto+Mono:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext">

<link rel="stylesheet" href="//cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.0/css/all.min.css">
  <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/animate.css@3.1.1/animate.min.css">

<script class="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Gemini","version":"8.0.1","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12},"copycode":false,"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"prism":false,"i18n":{"placeholder":"Searching...","empty":"We didn't find any results for the search: ${query}","hits_time":"${hits} results found in ${time} ms","hits":"${hits} results found"}};
  </script>

  <meta name="description" content="Abstract Understanding cell heterogeneity is the prerequisite for cell development and disease-related studies. Traditional strategies like marker-dependent cell sorting or single-cell RNA sequencing">
<meta property="og:type" content="article">
<meta property="og:title" content="Benchmarking Several Regression Models on Dissecting Cell Type Compositions in Brain Tissues">
<meta property="og:url" content="http://example.com/2020/07/08/dcBenchmark/index.html">
<meta property="og:site_name" content="Ao Huang&#39;s Blog">
<meta property="og:description" content="Abstract Understanding cell heterogeneity is the prerequisite for cell development and disease-related studies. Traditional strategies like marker-dependent cell sorting or single-cell RNA sequencing">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="http://example.com/2020/07/08/dcBenchmark/Slide1.jpeg">
<meta property="og:image" content="http://example.com/2020/07/08/dcBenchmark/Slide2.jpeg">
<meta property="og:image" content="http://example.com/2020/07/08/dcBenchmark/Slide3.jpeg">
<meta property="og:image" content="http://example.com/2020/07/08/dcBenchmark/Slide4.jpeg">
<meta property="article:published_time" content="2020-07-08T14:39:04.000Z">
<meta property="article:modified_time" content="2020-10-31T08:24:21.453Z">
<meta property="article:author" content="Ao Huang">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://example.com/2020/07/08/dcBenchmark/Slide1.jpeg">


<link rel="canonical" href="http://example.com/2020/07/08/dcBenchmark/">


<script class="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>Benchmarking Several Regression Models on Dissecting Cell Type Compositions in Brain Tissues | Ao Huang's Blog</title>
  






  <noscript>
  <style>
  body { margin-top: 2rem; }

  .use-motion .menu-item,
  .use-motion .sidebar,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header {
    visibility: visible;
  }

  .use-motion .header,
  .use-motion .site-brand-container .toggle,
  .use-motion .footer { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle,
  .use-motion .custom-logo-image {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line {
    transform: scaleX(1);
  }

  .search-pop-overlay, .sidebar-nav { display: none; }
  .sidebar-panel { display: block; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage" class="use-motion">
  <div class="headband"></div>

  <main class="main">
    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <h1 class="site-title">Ao Huang's Blog</h1>
      <i class="logo-line"></i>
    </a>
      <p class="site-subtitle" itemprop="description">Hello,world.</p>
      <img class="custom-logo-image" src="/images/imageTest.jpeg" alt="Ao Huang's Blog">
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>







</div>
        
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
  </div>

  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <div class="sidebar-panel-container">
        <!--noindex-->
        <section class="post-toc-wrap sidebar-panel">
            <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#abstract"><span class="nav-number">1.</span> <span class="nav-text"> Abstract</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#introduction"><span class="nav-number">2.</span> <span class="nav-text"> Introduction</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#classical-strategies-in-studying-cell-heterogeneity"><span class="nav-number">2.1.</span> <span class="nav-text"> Classical Strategies in Studying Cell Heterogeneity</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#rna-seq-and-scrna-seq"><span class="nav-number">2.2.</span> <span class="nav-text"> RNA-seq and scRNA-seq</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#deconvolution-and-its-application-in-brain-tissues"><span class="nav-number">2.3.</span> <span class="nav-text"> Deconvolution and its application in brain tissues</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#methods"><span class="nav-number">3.</span> <span class="nav-text"> Methods</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#rationale-for-deconvolution-in-estimating-cell-fractions"><span class="nav-number">3.1.</span> <span class="nav-text"> Rationale for deconvolution in estimating cell fractions</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#regression-based-deconvolution-in-this-study"><span class="nav-number">3.2.</span> <span class="nav-text"> Regression-based deconvolution in this study</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#data-summary-and-exploratory-analysis"><span class="nav-number">3.3.</span> <span class="nav-text"> Data summary and exploratory analysis</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#benchmark-strategy"><span class="nav-number">3.4.</span> <span class="nav-text"> Benchmark strategy</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#results"><span class="nav-number">4.</span> <span class="nav-text"> Results</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#svr-outperforms-all-three-least-squares-based-regression-methods-in-stability"><span class="nav-number">4.1.</span> <span class="nav-text"> SVR outperforms all three least squares-based regression methods in stability</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#svr-has-better-performances-in-enumerating-cell-subsets-with-closer-kinship"><span class="nav-number">4.2.</span> <span class="nav-text"> SVR has better performances in enumerating cell subsets with closer kinship</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#certain-cell-types-exhibit-systematic-bias-deconvolved-with-svr"><span class="nav-number">4.3.</span> <span class="nav-text"> Certain cell types exhibit systematic bias deconvolved with SVR</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#discussion"><span class="nav-number">5.</span> <span class="nav-text"> Discussion</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#supplementary-figures"><span class="nav-number">6.</span> <span class="nav-text"> Supplementary Figures</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#codes-availability"><span class="nav-number">7.</span> <span class="nav-text"> Codes Availability</span></a></li></ol></div>
        </section>
        <!--/noindex-->

        <section class="site-overview-wrap sidebar-panel">
          <div class="site-author site-overview-item animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">Ao Huang</p>
  <div class="site-description" itemprop="description">Father of Luo</div>
</div>
<div class="site-state-wrap site-overview-item animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives">
          <span class="site-state-item-count">1</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
        <span class="site-state-item-count">1</span>
        <span class="site-state-item-name">categories</span>
      </div>
  </nav>
</div>



        </section>
      </div>
    </div>
  </aside>
  <div class="sidebar-dimmer"></div>


    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>


    <div class="main-inner post posts-expand">
      

      

  


<div class="post-block">
  
  

  <article itemscope itemtype="http://schema.org/Article" class="post-content" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2020/07/08/dcBenchmark/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Ao Huang">
      <meta itemprop="description" content="Father of Luo">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Ao Huang's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          Benchmarking Several Regression Models on Dissecting Cell Type Compositions in Brain Tissues
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">Posted on</span>

      <time title="Created: 2020-07-08 22:39:04" itemprop="dateCreated datePublished" datetime="2020-07-08T22:39:04+08:00">2020-07-08</time>
    </span>
      <span class="post-meta-item">
        <span class="post-meta-item-icon">
          <i class="far fa-calendar-check"></i>
        </span>
        <span class="post-meta-item-text">Edited on</span>
        <time title="Modified: 2020-10-31 16:24:21" itemprop="dateModified" datetime="2020-10-31T16:24:21+08:00">2020-10-31</time>
      </span>
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-folder"></i>
      </span>
      <span class="post-meta-item-text">In</span>
        <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
          <a href="/categories/Research-Projects/" itemprop="url" rel="index"><span itemprop="name">Research Projects</span></a>
        </span>
    </span>

  
</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">
        <h3 id="abstract"><a class="markdownIt-Anchor" href="#abstract"></a> Abstract</h3>
<p>Understanding cell heterogeneity is the prerequisite for cell development and disease-related studies. Traditional strategies like marker-dependent cell sorting or single-cell RNA sequencing (scRNA-seq) have provided many insights in dissecting cell type-specific compositions and gene expression profiles in tissues, but there have been many drawbacks associating with the lack of precision and capability in cell isolations which limit their applications. Deconvolution emerges as a powerful tool in enumerating cell subsets and their specific gene expression patterns directly from bulk tissue transcriptomes by means of multivariate models without physical cell isolations. <strong><a id="more"></a></strong> Relevant pipelines and algorithms have been widely developed and applied in restoring cell type-specific transcriptomes in immune-infiltrated tissues, but there have been seldom reports on using deconvolution on brain tissues which are more heterogenous and complex. In this study, we test four regression methods, optimal least squares (OLS), LASSO, ridge and support vector regression (SVR) as deconvolution methods on raw scRNA-seq data of brain-specific cells. Through our designed benchmark strategy, we find that least squares-based regression models have better performance when the simulated bulk transcriptomes are cleaner and neater. When introduced with outliers, however, SVR exhibits more stable performance among the four methods in precisely predicting the fractions for each cell type. We then benchmark the influence of cell type phylogenetic kinship on SVR’s performance. To our surprise, SVR has better performance in enumerating cell types with closer kinship. We further target several cell types namely microglia, astrocytes, oligodendrocytes and their progenitors (OPC), excitatory neurons (subset 1, 2, 5b and 8) that have systematic bias through deconvolution by SVR. What contributes to these biases is still under investigations. In all, this study is an early endeavor to benchmark several regression methods’ performance on brain cells and to explore brain cell-specific transcriptomes data structures, which indicates that a deconvolution method specific for cell enumeration in brain tissues should be constructed for higher accuracy.</p>
<h3 id="introduction"><a class="markdownIt-Anchor" href="#introduction"></a> Introduction</h3>
<h4 id="classical-strategies-in-studying-cell-heterogeneity"><a class="markdownIt-Anchor" href="#classical-strategies-in-studying-cell-heterogeneity"></a> Classical Strategies in Studying Cell Heterogeneity</h4>
<p>Cell heterogeneity has been the principle research focus in basic biology ever since the dawn of molecular age. The decode of the heterogenous cell subsets morphologically, functionally and genetically has shed light on answering the most fundamental questions in cell differentiation, development and pathological process. Classical approaches in sorting and typing cells are through flow-cytometry and immunohistochemistry (IHC). Both of these methods depend on fluoresces-distinctive antibodies labeling, which can be visualized and analyzed for studying cell compositions and distribution. For these methods, however, priori knowledge on pre-selected protein or nucleic acid markers are needed to design cell type-specific markers, which are not suitable for exploratory analysis. Flow-cytometry and IHC study cell heterogeneity at a more macroscopic view and neglect the identification and quantification of cell subsets with more subtle differences.</p>
<h4 id="rna-seq-and-scrna-seq"><a class="markdownIt-Anchor" href="#rna-seq-and-scrna-seq"></a> RNA-seq and scRNA-seq</h4>
<p>The emergence of sequencing technology on transcriptomes brings examinations on cell heterogeneity to gene-expression-specific resolution. Transcriptomes measure messenger RNA (mRNA) abundance in a bulk tissue or even a single cell. mRNA are transcribed from DNA, and therefore by knowing mRNA compositions, cellular functions and stage of development can be deduced. All cells within an individual organism shares totally identical genomes, but their gene expression profiles (GEPs), partly and majorly represented by mRNA abundance, are very distinct temporally and spatially, which contribute to cell heterogeneity.</p>
<p>RNA sequencing (RNA-seq) technology is a prevalent strategy in quantifying mRNA profiles within biological tissues as GEPs. RNA-seq are even ameliorated to be done at single-cell level that unveil single cell-specific transcriptomes, abbreviated as single-cell RNA sequencing (scRNA-seq). Upon obtaining the scRNA-seq data from a bulk of tissue, each and every cell type can be identified based on their unique GEPs fingerprints. However, as powerful as scRNA-seq is, it has certain dilemmas limiting its applications:</p>
<ul>
<li>Large tissue samples and fixed clinical specimens cannot form single-cell suspensions, which is a prior step for scRNA-seq;</li>
<li>Transcriptomes data might be damaged when forming single-cell suspensions;</li>
<li>scRNA-seq is relatively expensive.</li>
</ul>
<p>In these scenarios, we can only obtain RNA-seq data which quantifies the bulk transcriptomes of tissues. How to obtain from bulk RNA-seq data the cell type-specific GEPs and compositions remains a challenging and unsettled question.</p>
<h4 id="deconvolution-and-its-application-in-brain-tissues"><a class="markdownIt-Anchor" href="#deconvolution-and-its-application-in-brain-tissues"></a> Deconvolution and its application in brain tissues</h4>
<p>In recent years, computational techniques and mathematical models have been developed assisting in enumerating cell types and compositions from a bulk tissue, deconvolution among them. Deconvolution has many interpretations in different disciplines and targeted questions, but for cell heterogeneity studies, deconvolution is explained as being capable of deducing cell type-specific GEPs from heterogenous tissues without physical cell isolation \cite{avila2018computational}. Deconvolution methods have been widely developed in dissecting tumor-infiltrating immune cells and achieve excellent performances \cite{abbas2009deconvolution, newman2015robust, newman2019determining}.</p>
<p>However, there have been seldom reports on using deconvolution methods in dissecting cell subsets in brain tissues. Frankly speaking, brain tissues is in more desperate needs for deconvolution in studying hetergeneity, because neuron cells have tremendous complex and baroque 3-d structures which are hard to disaggregate for scRNA-seq \cite{mckenzie2018brain} . Additionally, brain-specific cell compositions have even higher heterogeneity which requires more robust deconvolution method in dealing with noise and subtle GEPs differences.</p>
<p>There are two major goals in deconvolution: (1) enumerate the types of cells in bulk tissues and their corresponding proportions. (2) compute cell type-specific GEPs. In this study, we mainly focus on the first goal, because upon gathering the cell fractions information from a bulk tissue, cell-specific GEPs can be postulated more easily. We have tested four regression models, optimal least squares (OLS), least absolute shrinkage and selection operator (LASSO), ridge and support vector regression (SVR) as deconvolution strategies in brain single-cell transcriptomes data. By testing regression methods on pseudo bulk RNA-seq data generated from scRNA-seq data, we share insights in the performances and defects for current multivariate methods in brain cell-specific deconvolution as a preceded exploratory analysis for a more suitable deconvolution method developed for brain cell-specific analysis.</p>
<h3 id="methods"><a class="markdownIt-Anchor" href="#methods"></a> Methods</h3>
<h4 id="rationale-for-deconvolution-in-estimating-cell-fractions"><a class="markdownIt-Anchor" href="#rationale-for-deconvolution-in-estimating-cell-fractions"></a> Rationale for deconvolution in estimating cell fractions</h4>
<p>The basic assumption of applying deconvolution in cell subsets enumeration is that the expression level for a specific gene in a bulk tissue is the <em>linear combinations</em> of that gene expressed in each cell type within the studied bulk tissue (Figure 1a). It takes the form of linear system of equations that associates bulk GEPs, cell type-specific GEPs and fractions for each cell types:</p>
<p class='katex-block'><span class="katex-display"><span class="katex"><span class="katex-mathml"><math><semantics><mtable width="100%"><mtr><mtd width="50%"></mtd><mtd><mrow><msub><mi mathvariant="bold">Y</mi><mrow><mi>N</mi><mo>×</mo><mn>1</mn></mrow></msub><mo>=</mo><msub><mi mathvariant="bold">X</mi><mrow><mi>N</mi><mo>×</mo><mi>C</mi></mrow></msub><mo>⋅</mo><msub><mi>β</mi><mrow><mi>C</mi><mo>×</mo><mn>1</mn></mrow></msub></mrow></mtd><mtd width="50%"></mtd><mtd><mtext>(1)</mtext></mtd></mtr></mtable><annotation encoding="application/x-tex">\mathbf{Y}_{N\times1} = \mathbf{X}_{N\times C} \cdot \pmb{\beta}_{C\times1} \tag{1}
</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.894441em;vertical-align:-0.208331em;"></span><span class="mord"><span class="mord"><span class="mord mathbf" style="margin-right:0.02875em;">Y</span></span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.328331em;"><span style="top:-2.5500000000000003em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathdefault mtight" style="margin-right:0.10903em;">N</span><span class="mbin mtight">×</span><span class="mord mtight">1</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.208331em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.894441em;vertical-align:-0.208331em;"></span><span class="mord"><span class="mord"><span class="mord mathbf">X</span></span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.328331em;"><span style="top:-2.5500000000000003em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathdefault mtight" style="margin-right:0.10903em;">N</span><span class="mbin mtight">×</span><span class="mord mathdefault mtight" style="margin-right:0.07153em;">C</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.208331em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">⋅</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span></span><span class="base"><span class="strut" style="height:1.019133222222222em;vertical-align:-0.302471em;"></span><span class="mord"><span class="mord"><span class="mord"><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.69444em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="inner"><span class="mord"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.19444em;"><span></span></span></span></span></span><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.17221777777777778em;"></span><span class="inner"><span class="mord"><span class="mspace" style="margin-right:0.022222222222222223em;"></span><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3.0222222222222226em;"><span class="pstrut" style="height:3em;"></span><span class="mord text"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span><span class="mord"><span class="mspace" style="margin-right:0.044444444444444446em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.23419099999999998em;"><span style="top:-2.45586em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathdefault mtight" style="margin-right:0.07153em;">C</span><span class="mbin mtight">×</span><span class="mord mtight">1</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.302471em;"><span></span></span></span></span></span></span></span><span class="tag"><span class="strut" style="height:1.052471em;vertical-align:-0.302471em;"></span><span class="mord text"><span class="mord">(</span><span class="mord"><span class="mord">1</span></span><span class="mord">)</span></span></span></span></span></span></p>
<p>In equation 1, <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi mathvariant="bold">Y</mi></mrow><annotation encoding="application/x-tex">\mathbf{Y}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68611em;vertical-align:0em;"></span><span class="mord"><span class="mord mathbf" style="margin-right:0.02875em;">Y</span></span></span></span></span> is the bulk matrix representing the expression levels of <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>N</mi></mrow><annotation encoding="application/x-tex">N</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.10903em;">N</span></span></span></span> genes in a bulk tissue, <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi mathvariant="bold">X</mi></mrow><annotation encoding="application/x-tex">\mathbf{X}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68611em;vertical-align:0em;"></span><span class="mord"><span class="mord mathbf">X</span></span></span></span></span> is the reference matrix with cell type-specific GEPs of <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>C</mi></mrow><annotation encoding="application/x-tex">C</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.07153em;">C</span></span></span></span> types of cells and <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>β</mi></mrow><annotation encoding="application/x-tex">\pmb{\beta}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.911102222222222em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord"><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.69444em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="inner"><span class="mord"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.19444em;"><span></span></span></span></span></span><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.17221777777777778em;"></span><span class="inner"><span class="mord"><span class="mspace" style="margin-right:0.022222222222222223em;"></span><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3.0222222222222226em;"><span class="pstrut" style="height:3em;"></span><span class="mord text"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span><span class="mord"><span class="mspace" style="margin-right:0.044444444444444446em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span></span></span></span> is the fraction matrix containing <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>C</mi></mrow><annotation encoding="application/x-tex">C</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.07153em;">C</span></span></span></span> proportions corresponding to each cell types. <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi mathvariant="bold">Y</mi></mrow><annotation encoding="application/x-tex">\mathbf{Y}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68611em;vertical-align:0em;"></span><span class="mord"><span class="mord mathbf" style="margin-right:0.02875em;">Y</span></span></span></span></span> can be collected from RNA-seq data for a bulk tissue, <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi mathvariant="bold">X</mi></mrow><annotation encoding="application/x-tex">\mathbf{X}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68611em;vertical-align:0em;"></span><span class="mord"><span class="mord mathbf">X</span></span></span></span></span> can be roughly  assembled from scRNA-seq data of each cell type (do not necessarily have to come from the same tissue). Then, the simple goal of deconvolution is to estimates the <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>β</mi></mrow><annotation encoding="application/x-tex">\pmb{\beta}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.911102222222222em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord"><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.69444em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="inner"><span class="mord"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.19444em;"><span></span></span></span></span></span><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.17221777777777778em;"></span><span class="inner"><span class="mord"><span class="mspace" style="margin-right:0.022222222222222223em;"></span><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3.0222222222222226em;"><span class="pstrut" style="height:3em;"></span><span class="mord text"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span><span class="mord"><span class="mspace" style="margin-right:0.044444444444444446em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span></span></span></span> coefficients for a systems of linear equations, which represent cell fractions.</p>
<p>Notably, we often need to generate a signature matrix <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi mathvariant="bold">S</mi></mrow><annotation encoding="application/x-tex">\mathbf{S}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68611em;vertical-align:0em;"></span><span class="mord"><span class="mord mathbf">S</span></span></span></span></span> containing a subsets of marker genes <span class="katex"><span class="katex-mathml"><math><semantics><mrow><msup><mi>N</mi><mo mathvariant="normal">′</mo></msup></mrow><annotation encoding="application/x-tex">N&#x27;</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.751892em;vertical-align:0em;"></span><span class="mord"><span class="mord mathdefault" style="margin-right:0.10903em;">N</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.751892em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">′</span></span></span></span></span></span></span></span></span></span></span></span> from the reference matrix (Figure 1a). Because not all genes’ expression levels can serve as robust training sets in distinguishing different cell types. We set two thresholds in filtering the marker genes. A gene is considered as a marker gene if:</p>
<ul>
<li>Its expression level in at least one cell type is 5 times higher than the median expression level of all cell types;</li>
<li>The fold change between expression level in one cell type and mean expression level of all cell types converted to logarithm of base two is greater than 3, that is</li>
</ul>
<p class='katex-block'><span class="katex-display"><span class="katex"><span class="katex-mathml"><math><semantics><mrow><msub><mtext>log</mtext><mn>2</mn></msub><mo stretchy="false">(</mo><mrow><mi mathvariant="normal">F</mi><mi mathvariant="normal">C</mi></mrow><mo stretchy="false">)</mo><mo>=</mo><msub><mtext>log</mtext><mn>2</mn></msub><mo stretchy="false">(</mo><mtext>  </mtext><mfrac><mrow><mi>c</mi><mi>e</mi><mi>l</mi><mi>l</mi><mtext>  </mtext><mi>e</mi><mi>x</mi><mi>p</mi><mi>r</mi></mrow><mrow><mi>m</mi><mi>e</mi><mi>a</mi><mi>n</mi><mtext>  </mtext><mi>e</mi><mi>x</mi><mi>p</mi><mi>r</mi></mrow></mfrac><mtext>  </mtext><mo stretchy="false">)</mo><mo>&gt;</mo><mn>3.</mn></mrow><annotation encoding="application/x-tex">\textrm{log}_2(\mathrm{FC}) = \textrm{log}_2(\;\frac{cell\;expr}{mean\; expr}\;) &gt; 3.
</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em;"></span><span class="mord"><span class="mord text"><span class="mord textrm">log</span></span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.20696799999999996em;"><span style="top:-2.4558600000000004em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.24414em;"><span></span></span></span></span></span></span><span class="mopen">(</span><span class="mord"><span class="mord mathrm">F</span><span class="mord mathrm">C</span></span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:2.25188em;vertical-align:-0.8804400000000001em;"></span><span class="mord"><span class="mord text"><span class="mord textrm">log</span></span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.20696799999999996em;"><span style="top:-2.4558600000000004em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.24414em;"><span></span></span></span></span></span></span><span class="mopen">(</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mord"><span class="mopen nulldelimiter"></span><span class="mfrac"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.3714399999999998em;"><span style="top:-2.314em;"><span class="pstrut" style="height:3em;"></span><span class="mord"><span class="mord mathdefault">m</span><span class="mord mathdefault">e</span><span class="mord mathdefault">a</span><span class="mord mathdefault">n</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mord mathdefault">e</span><span class="mord mathdefault">x</span><span class="mord mathdefault">p</span><span class="mord mathdefault" style="margin-right:0.02778em;">r</span></span></span><span style="top:-3.23em;"><span class="pstrut" style="height:3em;"></span><span class="frac-line" style="border-bottom-width:0.04em;"></span></span><span style="top:-3.677em;"><span class="pstrut" style="height:3em;"></span><span class="mord"><span class="mord mathdefault">c</span><span class="mord mathdefault">e</span><span class="mord mathdefault" style="margin-right:0.01968em;">l</span><span class="mord mathdefault" style="margin-right:0.01968em;">l</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mord mathdefault">e</span><span class="mord mathdefault">x</span><span class="mord mathdefault">p</span><span class="mord mathdefault" style="margin-right:0.02778em;">r</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.8804400000000001em;"><span></span></span></span></span></span><span class="mclose nulldelimiter"></span></span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">&gt;</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:0.64444em;vertical-align:0em;"></span><span class="mord">3</span><span class="mord">.</span></span></span></span></span></p>
<p>Figure 1b presents an example in marker genes selection of one of the trails in the coming benchmarking. The aforesaid thresholds filter 2421 marker genes from the original 17176 genes, which give more distinct GEPs features for each cell types.</p>
<img src="/2020/07/08/dcBenchmark/Slide1.jpeg" class="">
<h4 id="regression-based-deconvolution-in-this-study"><a class="markdownIt-Anchor" href="#regression-based-deconvolution-in-this-study"></a> Regression-based deconvolution in this study</h4>
<p><strong>Simple linear regression.</strong> Since deconvolution in cell subsets enumeration is to fit a matrix equation, the first thought is to find a set of <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>β</mi></mrow><annotation encoding="application/x-tex">\beta</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span> that minimize the error sums of squares, that is the simple regression by OLS. Our trails are run with the ``lm&quot; function in R. Note that the coefficients estimated by OLS is forced to be non-negative and normalized to sum to one, because <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>β</mi></mrow><annotation encoding="application/x-tex">\pmb{\beta}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.911102222222222em;vertical-align:-0.19444em;"></span><span class="mord"><span class="mord"><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.69444em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="inner"><span class="mord"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.19444em;"><span></span></span></span></span></span><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.17221777777777778em;"></span><span class="inner"><span class="mord"><span class="mspace" style="margin-right:0.022222222222222223em;"></span><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3.0222222222222226em;"><span class="pstrut" style="height:3em;"></span><span class="mord text"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span><span class="mord"><span class="mspace" style="margin-right:0.044444444444444446em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span></span></span></span> means cell fractions in biological context.</p>
<p><strong>Penalized regression (LASSO and ridge).</strong> In real case analysis, however, bulk transcriptomes are not necessarily the linear combinations of cell type-specific GEPs, because for an totally unexplored bulk tissue, we can only estimate roughly of the types of cells in this tissue through means of flow-cytometry or IHC. Besides, scRNA-seq has its own bias that does not always provide accurate information on cell-type transcriptomes. These indicates that there would be quite some noise and disturbance in data structures that undermine deconvolution performances. Henceforth, we test LASSO and ridge regression with penalty in compromise of the disturbance, whose cost functions are depicted as</p>
<p class='katex-block'><span class="katex-display"><span class="katex"><span class="katex-mathml"><math><semantics><mtable rowspacing="0.15999999999999992em" columnalign="right center left" columnspacing="1em"><mtr><mtd><mstyle scriptlevel="0" displaystyle="false"><mrow><mi>ϕ</mi><mo stretchy="false">(</mo><mi>β</mi><mo stretchy="false">)</mo></mrow></mstyle></mtd><mtd><mstyle scriptlevel="0" displaystyle="false"><mo>=</mo></mstyle></mtd><mtd><mstyle scriptlevel="0" displaystyle="false"><mrow><mi mathvariant="normal">∥</mi><msup><mi>β</mi><mi>τ</mi></msup><mi mathvariant="bold">X</mi><mo>−</mo><mi mathvariant="bold">Y</mi><msup><mi mathvariant="normal">∥</mi><mn>2</mn></msup><mo>+</mo><mi>λ</mi><mi>p</mi><mo stretchy="false">(</mo><mi>β</mi><mo stretchy="false">)</mo></mrow></mstyle></mtd></mtr><mtr><mtd><mstyle scriptlevel="0" displaystyle="false"><mrow><mrow><mi mathvariant="normal">w</mi><mi mathvariant="normal">h</mi><mi mathvariant="normal">e</mi><mi mathvariant="normal">r</mi><mi mathvariant="normal">e</mi></mrow><mo separator="true">,</mo><mtext>  </mtext><mi>p</mi><mo stretchy="false">(</mo><mi>β</mi><mo stretchy="false">)</mo></mrow></mstyle></mtd><mtd><mstyle scriptlevel="0" displaystyle="false"><mo>=</mo></mstyle></mtd><mtd><mstyle scriptlevel="0" displaystyle="false"><mrow><mo fence="true">{</mo><mtable rowspacing="0.15999999999999992em" columnalign="left left" columnspacing="1em"><mtr><mtd><mstyle scriptlevel="0" displaystyle="false"><mrow><mi mathvariant="normal">∥</mi><mi>β</mi><mi mathvariant="normal">∥</mi></mrow></mstyle></mtd><mtd><mstyle scriptlevel="0" displaystyle="false"><mtext mathvariant="italic">LASSO,</mtext></mstyle></mtd></mtr><mtr><mtd><mstyle scriptlevel="0" displaystyle="false"><mrow><mi mathvariant="normal">∥</mi><mi>β</mi><msup><mi mathvariant="normal">∥</mi><mn>2</mn></msup></mrow></mstyle></mtd><mtd><mstyle scriptlevel="0" displaystyle="false"><mtext mathvariant="italic">Ridge.</mtext></mstyle></mtd></mtr></mtable></mrow></mstyle></mtd></mtr></mtable><annotation encoding="application/x-tex">\begin{array}{rcl}
		\phi(\pmb{\beta}) &amp;=&amp; \Vert \pmb{\beta}^{\tau}\mathbf{X} - \mathbf{Y} \Vert^2 + \lambda p(\pmb{\beta})  \\ 
	\mathrm{where}, \; p(\pmb{\beta}) &amp;=&amp; 
		\left \{
		\begin{array}{ll}
			\Vert \pmb{\beta} \Vert &amp; \mathrm{\textit{LASSO,}}  \\
			\Vert \pmb{\beta} \Vert^2 &amp; \mathrm{\textit{Ridge.}}
		\end{array}
		\right.	
	\end{array}
</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:3.6000300000000003em;vertical-align:-1.5500150000000001em;"></span><span class="mord"><span class="mtable"><span class="arraycolsep" style="width:0.5em;"></span><span class="col-align-r"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:2.050015em;"><span style="top:-4.6600150000000005em;"><span class="pstrut" style="height:3.45em;"></span><span class="mord"><span class="mord mathdefault">ϕ</span><span class="mopen">(</span><span class="mord"><span class="mord"><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.69444em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="inner"><span class="mord"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.19444em;"><span></span></span></span></span></span><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.17221777777777778em;"></span><span class="inner"><span class="mord"><span class="mspace" style="margin-right:0.022222222222222223em;"></span><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3.0222222222222226em;"><span class="pstrut" style="height:3em;"></span><span class="mord text"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span><span class="mord"><span class="mspace" style="margin-right:0.044444444444444446em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="mclose">)</span></span></span><span style="top:-2.850015em;"><span class="pstrut" style="height:3.45em;"></span><span class="mord"><span class="mord"><span class="mord mathrm" style="margin-right:0.01389em;">w</span><span class="mord mathrm">h</span><span class="mord mathrm">e</span><span class="mord mathrm">r</span><span class="mord mathrm">e</span></span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mord mathdefault">p</span><span class="mopen">(</span><span class="mord"><span class="mord"><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.69444em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="inner"><span class="mord"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.19444em;"><span></span></span></span></span></span><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.17221777777777778em;"></span><span class="inner"><span class="mord"><span class="mspace" style="margin-right:0.022222222222222223em;"></span><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3.0222222222222226em;"><span class="pstrut" style="height:3em;"></span><span class="mord text"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span><span class="mord"><span class="mspace" style="margin-right:0.044444444444444446em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="mclose">)</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:1.5500150000000001em;"><span></span></span></span></span></span><span class="arraycolsep" style="width:0.5em;"></span><span class="arraycolsep" style="width:0.5em;"></span><span class="col-align-c"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:2.050015em;"><span style="top:-4.6600150000000005em;"><span class="pstrut" style="height:3.45em;"></span><span class="mord"><span class="mrel">=</span></span></span><span style="top:-2.850015em;"><span class="pstrut" style="height:3.45em;"></span><span class="mord"><span class="mrel">=</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:1.5500150000000001em;"><span></span></span></span></span></span><span class="arraycolsep" style="width:0.5em;"></span><span class="arraycolsep" style="width:0.5em;"></span><span class="col-align-l"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:2.050015em;"><span style="top:-4.6600150000000005em;"><span class="pstrut" style="height:3.45em;"></span><span class="mord"><span class="mord">∥</span><span class="mord"><span class="mord"><span class="mord"><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.69444em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="inner"><span class="mord"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.19444em;"><span></span></span></span></span></span><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.17221777777777778em;"></span><span class="inner"><span class="mord"><span class="mspace" style="margin-right:0.022222222222222223em;"></span><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3.0222222222222226em;"><span class="pstrut" style="height:3em;"></span><span class="mord text"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span><span class="mord"><span class="mspace" style="margin-right:0.044444444444444446em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.7709542222222221em;"><span style="top:-3.1695622222222224em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathdefault mtight" style="margin-right:0.1132em;">τ</span></span></span></span></span></span></span></span></span><span class="mord"><span class="mord mathbf">X</span></span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">−</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mord"><span class="mord mathbf" style="margin-right:0.02875em;">Y</span></span><span class="mord"><span class="mord">∥</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mord mathdefault">λ</span><span class="mord mathdefault">p</span><span class="mopen">(</span><span class="mord"><span class="mord"><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.69444em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="inner"><span class="mord"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.19444em;"><span></span></span></span></span></span><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.17221777777777778em;"></span><span class="inner"><span class="mord"><span class="mspace" style="margin-right:0.022222222222222223em;"></span><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3.0222222222222226em;"><span class="pstrut" style="height:3em;"></span><span class="mord text"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span><span class="mord"><span class="mspace" style="margin-right:0.044444444444444446em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="mclose">)</span></span></span><span style="top:-2.850015em;"><span class="pstrut" style="height:3.45em;"></span><span class="mord"><span class="minner"><span class="mopen delimcenter" style="top:0em;"><span class="delimsizing size3">{</span></span><span class="mord"><span class="mtable"><span class="arraycolsep" style="width:0.5em;"></span><span class="col-align-l"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.45em;"><span style="top:-3.61em;"><span class="pstrut" style="height:3em;"></span><span class="mord"><span class="mord">∥</span><span class="mord"><span class="mord"><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.69444em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="inner"><span class="mord"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.19444em;"><span></span></span></span></span></span><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.17221777777777778em;"></span><span class="inner"><span class="mord"><span class="mspace" style="margin-right:0.022222222222222223em;"></span><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3.0222222222222226em;"><span class="pstrut" style="height:3em;"></span><span class="mord text"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span><span class="mord"><span class="mspace" style="margin-right:0.044444444444444446em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="mord">∥</span></span></span><span style="top:-2.4099999999999997em;"><span class="pstrut" style="height:3em;"></span><span class="mord"><span class="mord">∥</span><span class="mord"><span class="mord"><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.69444em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="inner"><span class="mord"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.19444em;"><span></span></span></span></span></span><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.17221777777777778em;"></span><span class="inner"><span class="mord"><span class="mspace" style="margin-right:0.022222222222222223em;"></span><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3.0222222222222226em;"><span class="pstrut" style="height:3em;"></span><span class="mord text"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span><span class="mord"><span class="mspace" style="margin-right:0.044444444444444446em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="mord"><span class="mord">∥</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.9500000000000004em;"><span></span></span></span></span></span><span class="arraycolsep" style="width:0.5em;"></span><span class="arraycolsep" style="width:0.5em;"></span><span class="col-align-l"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.45em;"><span style="top:-3.61em;"><span class="pstrut" style="height:3em;"></span><span class="mord"><span class="mord"><span class="mord text"><span class="mord textit">LASSO,</span></span></span></span></span><span style="top:-2.4099999999999997em;"><span class="pstrut" style="height:3em;"></span><span class="mord"><span class="mord"><span class="mord text"><span class="mord textit">Ridge.</span></span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.9500000000000004em;"><span></span></span></span></span></span><span class="arraycolsep" style="width:0.5em;"></span></span></span><span class="mclose nulldelimiter"></span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:1.5500150000000001em;"><span></span></span></span></span></span><span class="arraycolsep" style="width:0.5em;"></span></span></span></span></span></span></span></p>
<p>The algorithms is run using a R package <font face="Teletype">glmnet</font>. Note that the tuning parameter <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>λ</mi></mrow><annotation encoding="application/x-tex">\lambda</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.69444em;vertical-align:0em;"></span><span class="mord mathdefault">λ</span></span></span></span> is selected by ten-fold cross validation that minimize the error of sums of squares. Estimated coefficients are also forced to be non-negative and sum to one.</p>
<p><strong>Support vector regression (SVR).</strong> We extent support vector machine in linear form to the regression problem. Unlike the least squares-based regression methods, SVR inherited the properties of SVM which minimize only residuals that surpass a certain constant distance <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>ϵ</mi></mrow><annotation encoding="application/x-tex">\epsilon</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathdefault">ϵ</span></span></span></span>, whose cost function is</p>
<p class='katex-block'><span class="katex-display"><span class="katex"><span class="katex-mathml"><math><semantics><mtable width="100%"><mtr><mtd width="50%"></mtd><mtd><mtable rowspacing="0.15999999999999992em" columnalign="right center left" columnspacing="1em"><mtr><mtd><mstyle scriptlevel="0" displaystyle="false"><mrow><mi>ϕ</mi><mo stretchy="false">(</mo><mi>β</mi><mo stretchy="false">)</mo></mrow></mstyle></mtd><mtd><mstyle scriptlevel="0" displaystyle="false"><mo>=</mo></mstyle></mtd><mtd><mstyle scriptlevel="0" displaystyle="false"><mrow><mi>V</mi><mo stretchy="false">(</mo><msup><mi>β</mi><mi>τ</mi></msup><mi mathvariant="bold">X</mi><mo>−</mo><mi mathvariant="bold">Y</mi><mo stretchy="false">)</mo><mo>+</mo><mfrac><mi>λ</mi><mn>2</mn></mfrac><mi mathvariant="normal">∥</mi><mi>β</mi><msup><mi mathvariant="normal">∥</mi><mn>2</mn></msup></mrow></mstyle></mtd></mtr><mtr><mtd><mstyle scriptlevel="0" displaystyle="false"><mrow><mrow><mi mathvariant="normal">w</mi><mi mathvariant="normal">h</mi><mi mathvariant="normal">e</mi><mi mathvariant="normal">r</mi><mi mathvariant="normal">e</mi></mrow><mo separator="true">,</mo><mtext>  </mtext><msub><mi>V</mi><mi>ϵ</mi></msub><mo stretchy="false">(</mo><mi>r</mi><mo stretchy="false">)</mo></mrow></mstyle></mtd><mtd><mstyle scriptlevel="0" displaystyle="false"><mo>=</mo></mstyle></mtd><mtd><mstyle scriptlevel="0" displaystyle="false"><mrow><mo fence="true">{</mo><mtable rowspacing="0.15999999999999992em" columnalign="left left" columnspacing="1em"><mtr><mtd><mstyle scriptlevel="0" displaystyle="false"><mn>0</mn></mstyle></mtd><mtd><mstyle scriptlevel="0" displaystyle="false"><mrow><mrow><mi mathvariant="normal">i</mi><mi mathvariant="normal">f</mi></mrow><mi mathvariant="normal">∣</mi><mi>r</mi><mi mathvariant="normal">∣</mi><mo>&lt;</mo><mi>ϵ</mi><mo separator="true">,</mo></mrow></mstyle></mtd></mtr><mtr><mtd><mstyle scriptlevel="0" displaystyle="false"><mrow><mi mathvariant="normal">∣</mi><mi>r</mi><mi mathvariant="normal">∣</mi><mo>−</mo><mi>ϵ</mi></mrow></mstyle></mtd><mtd><mstyle scriptlevel="0" displaystyle="false"><mrow><mi mathvariant="normal">o</mi><mi mathvariant="normal">t</mi><mi mathvariant="normal">h</mi><mi mathvariant="normal">e</mi><mi mathvariant="normal">r</mi><mi mathvariant="normal">w</mi><mi mathvariant="normal">i</mi><mi mathvariant="normal">s</mi><mi mathvariant="normal">e</mi><mi mathvariant="normal">.</mi></mrow></mstyle></mtd></mtr></mtable></mrow></mstyle></mtd></mtr></mtable></mtd><mtd width="50%"></mtd><mtd><mtext>(2)</mtext></mtd></mtr></mtable><annotation encoding="application/x-tex">\begin{array}{rcl}
		\phi(\pmb{\beta}) &amp;=&amp; V( \pmb{\beta}^{\tau}\mathbf{X} - \mathbf{Y} ) + \frac{\lambda}{2} \Vert \pmb{\beta} \Vert^2  \\ 
	\mathrm{where}, \; V_{\epsilon}(r) &amp;=&amp; 
		\left \{
		\begin{array}{ll}
			0 &amp; \mathrm{if} \vert r \vert &lt; \epsilon, \\
			\vert r \vert - \epsilon &amp; \mathrm{otherwise.}
		\end{array}
		\right.	
	\end{array} \tag{2}
</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:3.6401380000000003em;vertical-align:-1.5700690000000002em;"></span><span class="mord"><span class="mtable"><span class="arraycolsep" style="width:0.5em;"></span><span class="col-align-r"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:2.070069em;"><span style="top:-4.6399610000000004em;"><span class="pstrut" style="height:3.45em;"></span><span class="mord"><span class="mord mathdefault">ϕ</span><span class="mopen">(</span><span class="mord"><span class="mord"><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.69444em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="inner"><span class="mord"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.19444em;"><span></span></span></span></span></span><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.17221777777777778em;"></span><span class="inner"><span class="mord"><span class="mspace" style="margin-right:0.022222222222222223em;"></span><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3.0222222222222226em;"><span class="pstrut" style="height:3em;"></span><span class="mord text"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span><span class="mord"><span class="mspace" style="margin-right:0.044444444444444446em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="mclose">)</span></span></span><span style="top:-2.829961em;"><span class="pstrut" style="height:3.45em;"></span><span class="mord"><span class="mord"><span class="mord mathrm" style="margin-right:0.01389em;">w</span><span class="mord mathrm">h</span><span class="mord mathrm">e</span><span class="mord mathrm">r</span><span class="mord mathrm">e</span></span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mord"><span class="mord mathdefault" style="margin-right:0.22222em;">V</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.151392em;"><span style="top:-2.5500000000000003em;margin-left:-0.22222em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathdefault mtight">ϵ</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em;"><span></span></span></span></span></span></span><span class="mopen">(</span><span class="mord mathdefault" style="margin-right:0.02778em;">r</span><span class="mclose">)</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:1.5700690000000002em;"><span></span></span></span></span></span><span class="arraycolsep" style="width:0.5em;"></span><span class="arraycolsep" style="width:0.5em;"></span><span class="col-align-c"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:2.070069em;"><span style="top:-4.6399610000000004em;"><span class="pstrut" style="height:3.45em;"></span><span class="mord"><span class="mrel">=</span></span></span><span style="top:-2.829961em;"><span class="pstrut" style="height:3.45em;"></span><span class="mord"><span class="mrel">=</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:1.5700690000000002em;"><span></span></span></span></span></span><span class="arraycolsep" style="width:0.5em;"></span><span class="arraycolsep" style="width:0.5em;"></span><span class="col-align-l"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:2.070069em;"><span style="top:-4.6399610000000004em;"><span class="pstrut" style="height:3.45em;"></span><span class="mord"><span class="mord mathdefault" style="margin-right:0.22222em;">V</span><span class="mopen">(</span><span class="mord"><span class="mord"><span class="mord"><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.69444em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="inner"><span class="mord"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.19444em;"><span></span></span></span></span></span><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.17221777777777778em;"></span><span class="inner"><span class="mord"><span class="mspace" style="margin-right:0.022222222222222223em;"></span><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3.0222222222222226em;"><span class="pstrut" style="height:3em;"></span><span class="mord text"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span><span class="mord"><span class="mspace" style="margin-right:0.044444444444444446em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.7709542222222221em;"><span style="top:-3.1695622222222224em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathdefault mtight" style="margin-right:0.1132em;">τ</span></span></span></span></span></span></span></span></span><span class="mord"><span class="mord mathbf">X</span></span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">−</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mord"><span class="mord mathbf" style="margin-right:0.02875em;">Y</span></span><span class="mclose">)</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">+</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mord"><span class="mopen nulldelimiter"></span><span class="mfrac"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.8801079999999999em;"><span style="top:-2.6550000000000002em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">2</span></span></span></span><span style="top:-3.23em;"><span class="pstrut" style="height:3em;"></span><span class="frac-line" style="border-bottom-width:0.04em;"></span></span><span style="top:-3.394em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathdefault mtight">λ</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.345em;"><span></span></span></span></span></span><span class="mclose nulldelimiter"></span></span><span class="mord">∥</span><span class="mord"><span class="mord"><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.69444em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.19444em;"></span><span class="inner"><span class="mord"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.19444em;"><span></span></span></span></span></span><span class="mord"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3em;"><span class="pstrut" style="height:3em;"></span><span class="rlap"><span class="strut" style="height:0.8888799999999999em;vertical-align:-0.17221777777777778em;"></span><span class="inner"><span class="mord"><span class="mspace" style="margin-right:0.022222222222222223em;"></span><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7166622222222221em;"><span style="top:-3.0222222222222226em;"><span class="pstrut" style="height:3em;"></span><span class="mord text"><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span></span><span class="fix"></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.17221777777777778em;"><span></span></span></span></span></span><span class="mord"><span class="mspace" style="margin-right:0.044444444444444446em;"></span><span class="mord mathdefault" style="margin-right:0.05278em;">β</span></span></span></span><span class="mord"><span class="mord">∥</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span><span style="top:-2.829961em;"><span class="pstrut" style="height:3.45em;"></span><span class="mord"><span class="minner"><span class="mopen delimcenter" style="top:0em;"><span class="delimsizing size3">{</span></span><span class="mord"><span class="mtable"><span class="arraycolsep" style="width:0.5em;"></span><span class="col-align-l"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.45em;"><span style="top:-3.61em;"><span class="pstrut" style="height:3em;"></span><span class="mord"><span class="mord">0</span></span></span><span style="top:-2.4099999999999997em;"><span class="pstrut" style="height:3em;"></span><span class="mord"><span class="mord">∣</span><span class="mord mathdefault" style="margin-right:0.02778em;">r</span><span class="mord">∣</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">−</span><span class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mord mathdefault">ϵ</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.9500000000000004em;"><span></span></span></span></span></span><span class="arraycolsep" style="width:0.5em;"></span><span class="arraycolsep" style="width:0.5em;"></span><span class="col-align-l"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.45em;"><span style="top:-3.61em;"><span class="pstrut" style="height:3em;"></span><span class="mord"><span class="mord"><span class="mord mathrm">i</span><span class="mord mathrm" style="margin-right:0.07778em;">f</span></span><span class="mord">∣</span><span class="mord mathdefault" style="margin-right:0.02778em;">r</span><span class="mord">∣</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">&lt;</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mord mathdefault">ϵ</span><span class="mpunct">,</span></span></span><span style="top:-2.4099999999999997em;"><span class="pstrut" style="height:3em;"></span><span class="mord"><span class="mord"><span class="mord mathrm">o</span><span class="mord mathrm">t</span><span class="mord mathrm">h</span><span class="mord mathrm">e</span><span class="mord mathrm">r</span><span class="mord mathrm" style="margin-right:0.01389em;">w</span><span class="mord mathrm">i</span><span class="mord mathrm">s</span><span class="mord mathrm">e</span><span class="mord mathrm">.</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.9500000000000004em;"><span></span></span></span></span></span><span class="arraycolsep" style="width:0.5em;"></span></span></span><span class="mclose nulldelimiter"></span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:1.5700690000000002em;"><span></span></span></span></span></span><span class="arraycolsep" style="width:0.5em;"></span></span></span></span><span class="tag"><span class="strut" style="height:3.6401380000000003em;vertical-align:-1.5700690000000002em;"></span><span class="mord text"><span class="mord">(</span><span class="mord"><span class="mord">2</span></span><span class="mord">)</span></span></span></span></span></span></p>
<p>From equation 2, SVR co-considers the selective error reduction and penalty function, which would minimize overfitting. In this study, we refer to a developed deconvolution package CIBERSORTx which uses <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>ν</mi></mrow><annotation encoding="application/x-tex">\nu</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.06366em;">ν</span></span></span></span>-support vector regression where the optimal <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>ν</mi></mrow><annotation encoding="application/x-tex">\nu</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.06366em;">ν</span></span></span></span>  have been selected for deconvolution in immune cells \cite{newman2015robust, newman2019determining}. The support vector regression process is run with the ``svm&quot; function in the R package <font face="Teletype">e1071</font> in CIBERSORTx. Since CIBERSORTx is designed as an intact pipeline for deconvolution including data normalization and building signature matrix etc., we conduct SVR directly through CIBERSORTx.</p>
<h4 id="data-summary-and-exploratory-analysis"><a class="markdownIt-Anchor" href="#data-summary-and-exploratory-analysis"></a> Data summary and exploratory analysis</h4>
<p>We test the above-mentioned four regression methods on brain cell-specific data, a 17176 (genes) by 27412 (cells) large matrix collected from <em>PsychENCODE</em> (<a target="_blank" rel="noopener" href="http://resource.psychencode.org/Datasets/Derived/SC_Decomp/DER-22_Single_cell_expression_raw_UMI.tsv">data availability</a>). In the dataset, each column represents a scRNA-seq data for a single cell and each row represents a gene expression values between cells. Each column of cell is a quantitative variable taken UMI (unique molecular identifier) as unit. The 27412 single cell GEPs was organized into 25 cell types and subsets by previous researches \cite{lake2018integrative} including astrocytes (Astro), endothelial cells (Endo), microglia cells (Microglia), Purkinje neurons (Per), oligodendrocytes (Oligo) and their progenitor cells (OPC), and excitatory (Ex) and inhibitory (In) neurons and their subsets (Figure 1c), which covers the major cell types in human brain.</p>
<p>We conduct t-distributed stochastic neighbor embedding (t-SNE) to reduce dimensions from 17176 to 2 to visualize cell type clustering based on their gene expression profiles (Figure 1d). t-SNE is a non-linear dimensionality reduction strategy that project high-dimensional data to low dimensional space while preserving high-dimensional clustering features. It defines point-to-point similarity by putting their distance in a t-distributed density function and reorganizes points based on their similarity scores in lower dimensions. t-SNE outperforms principal component analysis in preserving aggregation characteristics , the latter of which often creates overlaps between different clusters in spatial linear transformation to reduce dimensions. Figure 1d presents cell aggregates that shares similar GEPs. Generally, different cell types have been aggregated to different 2-d positions. But it is worth remarking that even for the same labeled cell types in the dataset like Ex1, Ex5b, OPC and Oligo, points are clustered to different segments, indicating that there are batch effects or temporal influences that result in distinct differences in scRNA-seq output, but these information have not been shed in the original dataset. Additionally, a confounding mixture is formed around the Per and Microglia labels. Notice that the scRNA-seq replicates for and Per cell type is very small (Figure 1c, 45 scRNA-seq samples), these data might not fully represents the GEPs of Per. We remove Per and preserve only the other 24 cell types in the following analysis.</p>
<p>For each cell type, we take the mean of their single cell expression profiles as the GEPs of that particular cell types. With the obtained cell type-specific GEPs as variables, we construct an agglomerative cluster tree using ward’s minimum variance (ward.D2) linkage method, which manifests the kinship between cell types (Figure 1e). Cell types are classified into seven groups based on their distance of relations.</p>
<h4 id="benchmark-strategy"><a class="markdownIt-Anchor" href="#benchmark-strategy"></a> Benchmark strategy</h4>
<p>We set the following benchmark strategy to compare different regression models on enumerating brain cells bulk transcriptomes (Figure 2a). We first decide the number of cell types to be deconvolved (the value of <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>C</mi></mrow><annotation encoding="application/x-tex">C</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.07153em;">C</span></span></span></span> in equation 1 and <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>C</mi></mrow><annotation encoding="application/x-tex">C</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.07153em;">C</span></span></span></span> in Figure 2a) and the number of scRNA-seq samples per cell type <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>R</mi></mrow><annotation encoding="application/x-tex">R</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.00773em;">R</span></span></span></span> to form the reference matrix <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi mathvariant="bold">M</mi></mrow><annotation encoding="application/x-tex">\mathbf{M}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68611em;vertical-align:0em;"></span><span class="mord"><span class="mord mathbf">M</span></span></span></span></span>. In  <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi mathvariant="bold">M</mi></mrow><annotation encoding="application/x-tex">\mathbf{M}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68611em;vertical-align:0em;"></span><span class="mord"><span class="mord mathbf">M</span></span></span></span></span>, <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>R</mi></mrow><annotation encoding="application/x-tex">R</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.00773em;">R</span></span></span></span> scRNA-seq samples are summed and divide by <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>R</mi></mrow><annotation encoding="application/x-tex">R</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.00773em;">R</span></span></span></span> (mean scRNA-seq GEPs) to represent cell type-specific GEPs. Signature matrix <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi mathvariant="bold">S</mi></mrow><annotation encoding="application/x-tex">\mathbf{S}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68611em;vertical-align:0em;"></span><span class="mord"><span class="mord mathbf">S</span></span></span></span></span> is filtered as mentioned above.</p>
<p>Once the value of <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>C</mi></mrow><annotation encoding="application/x-tex">C</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.07153em;">C</span></span></span></span> is determined, we set different sampling methods to select cell types with designed gradient of kinship closeness. Referencing Figure 1e, cell types are categorized into seven groups. Sampling method works like this: for method 1, <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>C</mi></mrow><annotation encoding="application/x-tex">C</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.07153em;">C</span></span></span></span> types of cell are randomly selected from the 24 cell types; for method 2, a cell type is randomly selected from the joint group of group 1 and group 2 and a second cell type is randomly selected from group 3, and another one from group 4, and another one from group 5, and another one from group 6 and group 7; the following methods also (Figure 2a Sampling Methods). These strategies are more suitable for sampling for five cell types, which is exactly the value we assign to <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>C</mi></mrow><annotation encoding="application/x-tex">C</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.07153em;">C</span></span></span></span> value in the following analysis.</p>
<p>Since we have only the single-cell transcriptomes data of brain cells, we  generate pseudo bulk transcriptomes, denoted as <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi mathvariant="bold">B</mi></mrow><annotation encoding="application/x-tex">\mathbf{B}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68611em;vertical-align:0em;"></span><span class="mord"><span class="mord mathbf">B</span></span></span></span></span> by randomly assigning cell fractions <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>F</mi></mrow><annotation encoding="application/x-tex">F</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.13889em;">F</span></span></span></span> to pre-chosen cell type-specific GEPs and sum them. To simulate analysis in real-time situation where the bulk transcriptomes contain noise that could not be fully interpreted by the reference matrix, we introduce normally distributed random error <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>ϵ</mi><mo>∼</mo><mi mathvariant="script">N</mi><mo stretchy="false">(</mo><mn>0</mn><mo separator="true">,</mo><msup><mi>σ</mi><mn>2</mn></msup><mo stretchy="false">)</mo></mrow><annotation encoding="application/x-tex">\epsilon \sim \mathcal{N}(0,\sigma^2)</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathdefault">ϵ</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">∼</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:1.064108em;vertical-align:-0.25em;"></span><span class="mord"><span class="mord mathcal" style="margin-right:0.14736em;">N</span></span><span class="mopen">(</span><span class="mord">0</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord"><span class="mord mathdefault" style="margin-right:0.03588em;">σ</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span><span class="mclose">)</span></span></span></span> to <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi mathvariant="bold">B</mi></mrow><annotation encoding="application/x-tex">\mathbf{B}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68611em;vertical-align:0em;"></span><span class="mord"><span class="mord mathbf">B</span></span></span></span></span>. We even introduce outliers to randomly-selected genes in several trails to benchmark the stability between models in dealing with strong disturbance.</p>
<p>The prepared <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi mathvariant="bold">S</mi></mrow><annotation encoding="application/x-tex">\mathbf{S}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68611em;vertical-align:0em;"></span><span class="mord"><span class="mord mathbf">S</span></span></span></span></span> and <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi mathvariant="bold">B</mi></mrow><annotation encoding="application/x-tex">\mathbf{B}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68611em;vertical-align:0em;"></span><span class="mord"><span class="mord mathbf">B</span></span></span></span></span> are put to training that produce a set of cell fractions predictions <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>P</mi></mrow><annotation encoding="application/x-tex">P</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.13889em;">P</span></span></span></span>, and a model is considered a better model if the pearson coefficients between the <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>F</mi></mrow><annotation encoding="application/x-tex">F</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.13889em;">F</span></span></span></span> and <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>P</mi></mrow><annotation encoding="application/x-tex">P</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.13889em;">P</span></span></span></span> is closer to one and root-mean square error (RMSE) closer to zero. This whole process from random sampling to calculating the performance scores is repeated <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>N</mi></mrow><annotation encoding="application/x-tex">N</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.10903em;">N</span></span></span></span> times for higher statistical confidence.</p>
<img src="/2020/07/08/dcBenchmark/Slide2.jpeg" class="">
<h3 id="results"><a class="markdownIt-Anchor" href="#results"></a> Results</h3>
<h4 id="svr-outperforms-all-three-least-squares-based-regression-methods-in-stability"><a class="markdownIt-Anchor" href="#svr-outperforms-all-three-least-squares-based-regression-methods-in-stability"></a> SVR outperforms all three least squares-based regression methods in stability</h4>
<p>Consulting the benchmark strategy in Figure 2a, we first set <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>C</mi></mrow><annotation encoding="application/x-tex">C</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.07153em;">C</span></span></span></span> to 5, <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>R</mi></mrow><annotation encoding="application/x-tex">R</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.00773em;">R</span></span></span></span> to 50 <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>N</mi></mrow><annotation encoding="application/x-tex">N</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.10903em;">N</span></span></span></span> to 100, and adopt method 1 in cell types sampling to benchmark the deconvolution performance of four regression models OLS, LASSO, ridge and SVR in restoring cell fractions. When introduced with the random error <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>ϵ</mi><mo>∼</mo><mi mathvariant="script">N</mi><mo stretchy="false">(</mo><mn>0</mn><mo separator="true">,</mo><mn>3</mn><msup><mn>0</mn><mn>2</mn></msup><mo stretchy="false">)</mo></mrow><annotation encoding="application/x-tex">\epsilon \sim \mathcal{N}(0,30^2)</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathdefault">ϵ</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span><span class="mrel">∼</span><span class="mspace" style="margin-right:0.2777777777777778em;"></span></span><span class="base"><span class="strut" style="height:1.064108em;vertical-align:-0.25em;"></span><span class="mord"><span class="mord mathcal" style="margin-right:0.14736em;">N</span></span><span class="mopen">(</span><span class="mord">0</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord">3</span><span class="mord"><span class="mord">0</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span><span class="mclose">)</span></span></span></span> in the bulk transcriptomes, OLS has the highest average pearson coefficients and lowest RMSE among the four methods (Figure 2b upper panel). The distribution of OLS’s pearson coefficents and RMSE per turn is also the most concentrated (Figure 2c right panel). It is not hard to account for such good performance OLS embodies under these set parameters, because the pseudo RNA transcriptomes are merely the linear combinations of each cell type GEPs with normally distributed error, which     would be easily neutralized in large numbers of repetition experiments. The penalized regression methods exhibit second best performance in this case. The choice of the tuning parameter <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>λ</mi></mrow><annotation encoding="application/x-tex">\lambda</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.69444em;vertical-align:0em;"></span><span class="mord mathdefault">λ</span></span></span></span> in both LASSO and ridge through ten-fold cross validation is very small, indicating that the penalty term take little effects in compromising normally distributed noise error (Supplementary Figure a).</p>
<p>We further introduce not only the random error but also outliers (1000 + <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi mathvariant="script">N</mi><mo stretchy="false">(</mo><mn>0</mn><mo separator="true">,</mo><mn>10</mn><msup><mn>0</mn><mn>2</mn></msup><mo stretchy="false">)</mo></mrow><annotation encoding="application/x-tex">\mathcal{N}(0,100^2)</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1.064108em;vertical-align:-0.25em;"></span><span class="mord"><span class="mord mathcal" style="margin-right:0.14736em;">N</span></span><span class="mopen">(</span><span class="mord">0</span><span class="mpunct">,</span><span class="mspace" style="margin-right:0.16666666666666666em;"></span><span class="mord">1</span><span class="mord">0</span><span class="mord"><span class="mord">0</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.8141079999999999em;"><span style="top:-3.063em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span><span class="mclose">)</span></span></span></span>) in 0.5% of randomly selected genes among the entire 17176 genes when constructing pseudo bulk transcriptomes. The SVR performed on CIBERSORTx has the optimal performances measured by average pearson coefficients and RMSE values (Figure 2b lower panel). The distribution of the pearson and RMSE values of SVR are also the most centralized and have fewer outliers (Figure 2c left panel). The median of the pearson value in OLS is higher than that of SVR, but its average pearson value is 0.07 lower than SVR, indicating that the OLS results of deconvolution is more sparsely distributed and not as stable as SVR. Besides, ridge regression outperforms LASSO and OLS in this scenario. Note that we have not been introduced penalized factor as a mean of feature selection, because each coefficient in the model have its biological meanings. Instead, we want the penalized term to neutralize distinctive deviations in the response variable. Ridge regression has realized this effect under this circumstances, its <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>λ</mi></mrow><annotation encoding="application/x-tex">\lambda</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.69444em;vertical-align:0em;"></span><span class="mord mathdefault">λ</span></span></span></span> being selected significantly higher than the <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>λ</mi></mrow><annotation encoding="application/x-tex">\lambda</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.69444em;vertical-align:0em;"></span><span class="mord mathdefault">λ</span></span></span></span> selected in LASSO (Supplementary Figure a).</p>
<p>Meanwhile, by comparing vertically in the scatterplots, scatter pattern in SVR become slightly more sparse after being introduced the outliers, but that of the least squares-based regression methods change significantly (Figure 1b). It can be concluded that SVR is a more robust and stable deconvolution method in dealing with data with more noise and disturbance, while OLS method is more suitable for analysis towards more confident and neat dataset. But the existence of outliers is more similar to the actual cases in practicing deconvolution on heterogenous brain-specific cells, where the types of cells within a bulk tissue are unable to be fully identified.</p>
<h4 id="svr-has-better-performances-in-enumerating-cell-subsets-with-closer-kinship"><a class="markdownIt-Anchor" href="#svr-has-better-performances-in-enumerating-cell-subsets-with-closer-kinship"></a> SVR has better performances in enumerating cell subsets with closer kinship</h4>
<p>Since SVR is the most stable in brain cell-type specific deconvolution, we intend to benchmark whether SVR is susceptible to cell kinship. We separately conduct benchmarking with five different sampling methods as shown in Figure 1a, with <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>C</mi></mrow><annotation encoding="application/x-tex">C</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.07153em;">C</span></span></span></span> to 5, <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>R</mi></mrow><annotation encoding="application/x-tex">R</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.00773em;">R</span></span></span></span> to 50 <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>N</mi></mrow><annotation encoding="application/x-tex">N</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.10903em;">N</span></span></span></span> to 100. Surprisingly, closer the kinship of the selected cell types, higher the accuracy SVR exhibits in deconvolution (Figure 3a). Method 5 merely samples from group 6 the subsets of inhibitory cells which is considered to have the closest relationship by tree clustering (Figure 1e). We set the loop numbers <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>N</mi></mrow><annotation encoding="application/x-tex">N</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.10903em;">N</span></span></span></span> from 100 to 2 to increase the random effects, but prediction accuracy measured by pearson and RMSE values shows similar tendency. It is worth noticing that the output of method 2 performs the most poorly with the lowest pearson value and highest RMSE and dispersed distributions (Figure 3c), even worse than the randomly sampled trails (method 1). Besides, the pairwise scatterplots of the pearson coefficients between the five methods show similar points clustering patterns, which indicates that there are a certain set of cell types that can be precisely estimated through SVR and another set of cell types that cannot (Supplementary Figure b). We reckon that the poor performance presents in method 2 can be attributed to a list of certain cell types whose assigned fractions cannot be precisely predicted by the SVR deconvolution method.</p>
<h4 id="certain-cell-types-exhibit-systematic-bias-deconvolved-with-svr"><a class="markdownIt-Anchor" href="#certain-cell-types-exhibit-systematic-bias-deconvolved-with-svr"></a> Certain cell types exhibit systematic bias deconvolved with SVR</h4>
<p>To test our assumption, instead of sampling cell types, we eliminate the random effect of the times a certain cell type can be selected during loops by fixing two groups of cell types with different closeness of kinship based on the classification trees (Figure 1e). The close group is comprised of In7, In8, In1c,In4a and In6a which are phylogenetically more closely related. The distant group contains Ex1, Endo, Microglia, Ex3e and In8 which are more distant. Since different linkage methods might result in totally different classification trees, we construct four classification trees using different linkage methods and find that the close group and distant group maintains their closeness of kinship in no matter what types of trees (Supplementary Figure c). In accordance with our assumption, estimations towards cell types in the close group performs better using SVR in deconvolution (Figure 3d). Surprisingly, by color-labeling the five selected cell types in the distant group, we find that the estimations of two cell types Ex1 and Microglia systematically deviates from the ``\textit{y = x}&quot; reference line. By removing the deviated points in Ex1 and Microglia from the scatterplot, we can calculate a trimmed average pearson coefficient which is improved from 0.6962 to 0.9092.</p>
<p>Ex1 and Microglia are from group 2 and group 5 respectively (Figure 3e right panel), indicating that there might be systematic biases for cell types deconvolved with SVR in these groups. We color-label that previous scatterplot that are produced with sampling method 2 and 100 loops in Figure 3e, and find that Ex2 in group 1, Ex1, Ex5b and Ex8 in group 2, Astro, Oligo, Microglia, OPC in group 5 have systematic biases when analyzed with the SVR (Figure 3e). The fractions of Ex1, Ex2 and Ex5b from group 1 and group 2 tend to be overestimated, while Astro, Oligo, Microglia and OPC in group 5 tend to be underestimated. These obviously manifests that the SVR deconvolution algorithms have systematic biases towards estimating certain brain cell types fractions.</p>
<p>Additionally, the values of <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>C</mi></mrow><annotation encoding="application/x-tex">C</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.07153em;">C</span></span></span></span> and <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>R</mi></mrow><annotation encoding="application/x-tex">R</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.00773em;">R</span></span></span></span>'s influence on SVR performance have also been benchmarked. The rise of <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>C</mi></mrow><annotation encoding="application/x-tex">C</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.07153em;">C</span></span></span></span> from 5 to 10 cell types undermines SVR performance (Supplementary Figure d). This is due to the curse of dimensionality which makes determining hyperplane boundaries more difficult for SVR at higher dimension. Note that the value of <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>R</mi></mrow><annotation encoding="application/x-tex">R</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.00773em;">R</span></span></span></span> from 50 to 180 have seldom influence on SVR’s performance but from 50 to 10 presents significant drawback (Supplementary Figure e). This partly explains our choice of <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>R</mi></mrow><annotation encoding="application/x-tex">R</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.68333em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.00773em;">R</span></span></span></span> in benchmarking which both considers accuracy and reduce computational cost.</p>
<img src="/2020/07/08/dcBenchmark/Slide3.jpeg" class="">
<h3 id="discussion"><a class="markdownIt-Anchor" href="#discussion"></a> Discussion</h3>
<p>Next generation sequencing has brought tremendous amounts of data in biological research which enables the study of cell heterogeneity through computational and mathematical approaches without actual physical cell isolations. Upon fitting a system of linear equations, deconvolution methods achieve satisfied performance in dissecting cell subsets in tumor and immunological tissues. In this study, we benchmark four regression strategies as deconvolution methods in restoring cell type-specific gene expression patterns on brain tissues where deconvolution methods are seldom applied. Note that we are not taking gene as predictors like what multivariate analysis towards GEPs normally does in deconvolution, but as training samples, because our purpose is to fit the coefficients for each cell type (variable) rather than taking them as qualitative responses.</p>
<p>Generally speaking, deconvolution with OLS has better performance with only normally distributed error introduced, which applies to the scenario where the prior knowledge to the content of bulk tissues are more sufficient. For more complicated and heterogenous tissues in lack of the intact knowledge of cell type compositions, SVR has better performances in dealing with more intense disturbance in a dataset. This is because simple regression is very sensitive to outliers in the training data while SVR provides more sparse solution to the regression model that are more tolerant towards outliers \cite{newman2015robust,scholkopf2000new}. For the penalized regression models, ridge performs better than the LASSO. This can be attributed to the difference in the cost function (equation 2), where in LASSO the estimates can be compressed to one while in ridge they cannot. This is in accordance to previous benchmarkings that coefficients predicted by LASSO tend to have larger biases \cite{izenman2008modern}.</p>
<p>The study of cell phylogenetic kinship on SVR deconvolution accuracy produce the most interesting results that closer the kinship of cell types, better the model performs. This is in collision with our previous understanding that more subtle the differences between GEPs, harder it is for deconvolution models to tell apart these cell types. With suspicions, we further discover that there exists systematic bias for certain cell types when applying SVR. By examining the cluster tree, we find that these bias cells are mainly located in groups with more distant kinship. But whether such bias are resulted from phylogenetic relations or the peculiar transcriptomes these bias cells have, or even the inveracity of the scRNA-seq data remains unanswered. Because from the t-SNE plot of Figure 1e, we can see that even though the selected cell types in the close group are closely related, but they have seldom overlaps in t-SNE plot, while for more distant related cell types (Figure 3e), their features after dimension reduction still mix with each another, especially for Microglia. These might be due from unexplored gene expression features among these cells or un-removed batch effects that influence the building cell type-specific GEPs from scRNA-seq. Additionally, the developed CIBERSORTx set <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>ν</mi></mrow><annotation encoding="application/x-tex">\nu</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.06366em;">ν</span></span></span></span> to be chosen from {0.25, 0.5, 0.75} which are found to perform the best in deconvolution in immune cells \cite{3}. We may try different <span class="katex"><span class="katex-mathml"><math><semantics><mrow><mi>ν</mi></mrow><annotation encoding="application/x-tex">\nu</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathdefault" style="margin-right:0.06366em;">ν</span></span></span></span> values in further analysis to achieve optimal performance for SVR in brain.</p>
<p>Still, deconvolution method is a promising computational tool to study cell heterogeneity. We anticipate that the systematic biases in brain cells be refined for more accurate enumeration of cell subsets and their unique expression profiles from brain-specific transcriptomes.</p>
<h3 id="supplementary-figures"><a class="markdownIt-Anchor" href="#supplementary-figures"></a> Supplementary Figures</h3>
<img src="/2020/07/08/dcBenchmark/Slide4.jpeg" class="">
<h3 id="codes-availability"><a class="markdownIt-Anchor" href="#codes-availability"></a> Codes Availability</h3>
<p><a href="codeAvailability.pdf">Link</a></p>

    </div>

    
    
    

    <footer class="post-footer">

        

    </footer>
  </article>
</div>






      

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      const activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      const commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

    </div>
  </main>

  <footer class="footer">
    <div class="footer-inner">
      

      

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Ao Huang</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.js.org/" class="theme-link" rel="noopener" target="_blank">NexT.Gemini</a>
  </div>

    </div>
  </footer>

  
  <script src="//cdn.jsdelivr.net/npm/animejs@3.2.0/lib/anime.min.js"></script>
<script src="/js/utils.js"></script><script src="/js/motion.js"></script><script src="/js/next-boot.js"></script>

  


















  








  

  
      <script>
  if (typeof MathJax === 'undefined') {
    window.MathJax = {
      tex: {
        inlineMath: {'[+]': [['$', '$']]},
        tags: 'ams'
      },
      options: {
        renderActions: {
          findScript: [10, doc => {
            document.querySelectorAll('script[type^="math/tex"]').forEach(node => {
              const display = !!node.type.match(/; *mode=display/);
              const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display);
              const text = document.createTextNode('');
              node.parentNode.replaceChild(text, node);
              math.start = {node: text, delim: '', n: 0};
              math.end = {node: text, delim: '', n: 0};
              doc.math.push(math);
            });
          }, '', false],
          insertedScript: [200, () => {
            document.querySelectorAll('mjx-container').forEach(node => {
              const target = node.parentNode;
              if (target.nodeName.toLowerCase() === 'li') {
                target.parentNode.classList.add('has-jax');
              }
            });
          }, '', false]
        }
      }
    };
    const script = document.createElement('script');
    script.src = '//cdn.jsdelivr.net/npm/mathjax@3.1.2/es5/tex-mml-chtml.js';
    script.defer = true;
    document.head.appendChild(script);
  } else {
    MathJax.startup.document.state(0);
    MathJax.typesetClear();
    MathJax.texReset();
    MathJax.typeset();
  }
</script>

    

  

</body>
</html>
